diff --git a/feeds_comment_processor.install b/feeds_comment_processor.install
index 2f655aa..7f78902 100644
--- a/feeds_comment_processor.install
+++ b/feeds_comment_processor.install
@@ -32,6 +32,7 @@ function feeds_comment_processor_schema() {
         'unsigned' => TRUE,
         'not null' => TRUE,
         'description' => "Node id of the owner feed, if available.",
+        'default' => 0,
       ),
       'imported' => array(
         'type' => 'int',
