diff --git a/sites/all/modules/contrib/clientside_validation/clientside_validation.js b/sites/all/modules/contrib/clientside_validation/clientside_validation.js
index dc6f86b..80382ad 100644
--- a/sites/all/modules/contrib/clientside_validation/clientside_validation.js
+++ b/sites/all/modules/contrib/clientside_validation/clientside_validation.js
@@ -1481,12 +1481,14 @@
         if (Drupal.ajax.hasOwnProperty(ajax_el) && typeof Drupal.ajax[ajax_el] !== 'undefined') {
           var $ajax_el = jQuery(Drupal.ajax[ajax_el].element);
           var ajax_form = $ajax_el.is('form') ? $ajax_el.attr('id') : $ajax_el.closest('form').attr('id');
-          var change_ajax = true;
-          if (typeof Drupal.myClientsideValidation.forms[ajax_form] !== 'undefined') {
-            change_ajax = Boolean(parseInt(Drupal.myClientsideValidation.forms[ajax_form].general.validateBeforeAjax, 10));
-          }
-          if (!$ajax_el.hasClass('cancel') && change_ajax) {
-            changeAjax(ajax_el);
+          if(ajax_form !== undefined){
+            var change_ajax = true;
+            if (typeof Drupal.myClientsideValidation.forms[ajax_form] !== 'undefined') {
+              change_ajax = Boolean(parseInt(Drupal.myClientsideValidation.forms[ajax_form].general.validateBeforeAjax, 10));
+            }
+            if (!$ajax_el.hasClass('cancel') && change_ajax) {
+              changeAjax(ajax_el);
+            }
           }
         }
       }
