diff --git a/search_api_db.install b/search_api_db.install
index d073dda..80db039 100644
--- a/search_api_db.install
+++ b/search_api_db.install
@@ -65,3 +65,56 @@ function search_api_db_update_7101() {
       ->execute();
   }
 }
+
+/**
+ * Implements hook_update_N().
+ *
+ * Solves index problems with umlauts, accented characters, etc. issue http://drupal.org/node/1144620.
+ */
+function search_api_db_update_7102() {
+  global $databases;
+  $search_api_tables = db_query("show tables LIKE '%search_api_db%';")->fetchCol();
+
+  $database = $databases['default']['default'];
+  $db_driver = $database['driver'];
+  $db_name = $database['database'];
+
+  switch ($db_driver) {
+    case 'mysql':
+      foreach ($search_api_tables as $table) {
+        db_query("ALTER TABLE {$table} CONVERT TO CHARACTER SET 'utf8' COLLATE 'utf8_bin'")
+        ->execute();
+      }
+      break;
+    case 'oracle':
+      // TODO: Check the character encoding and collation and change if required.
+      // e.g. db_query("ALTER DATABASE [$database] CHARACTER SET utf8_unicode_ci");
+      break;
+
+    case 'pgsql':
+      // PostgreSQL requires UTF8 from /includes/database/pgsql/install.inc
+      // checkEncoding() function, so no change should be necessary.
+      break;
+
+    case 'sqlite':
+      // TODO: The ALTER statement cannot change the collation, so a new table
+      // would have to be created with the correct collations on certain
+      // columns, the data moved, the old table dropped, and then the new table
+      // renamed to the old table's name.
+      break;
+
+    case 'sqlsrv':
+      // TODO: Do a test to ensure the columns of interest are unicode (nchar,
+      // nvarchar, ntext) and that the collation is accent-sensitive.
+
+      // The ALTER statment can only be applied to the database or to a column.
+      // Should the user be warned, should the database collation be
+      // changed, and/or should just some columns change?
+
+      // e.g. drupal_set_message(t("The database collation should be unicode case-insensitive accent-sensitive (e.g. SQL_Latin1_General_CP1_CI_AS)."));
+      // e.g. db_query("ALTER DATABASE [$db_name] COLLATE SQL_Latin1_General_CP1_CI_AS");
+      // e.g. db_query("ALTER TABLE {$table} ALTER COLUMN [$column] nvarchar(50) COLLATE SQL_Latin1_General_CP1_CI_AS NOT NULL");
+
+      break;
+  }
+}
\ No newline at end of file
diff --git a/service.inc b/service.inc
index 6b6e7f6..0df0164 100644
--- a/service.inc
+++ b/service.inc
@@ -192,6 +192,24 @@ class SearchApiDbService extends SearchApiAbstractService {
 
     $set = $this->setDb();
     db_create_table($name, $table);
+    
+    //Some DBMS's will need a character encoding and collation set.
+    global $databases;
+    $database = $databases['default']['default'];
+    $db_driver = $database['driver'];
+    $db_name = $database['database'];
+    
+    switch ($db_driver) {
+      case 'mysql':
+        db_query("ALTER TABLE {$name} CONVERT TO CHARACTER SET 'utf8' COLLATE 'utf8_bin'")->execute();
+        break;
+      case 'oracle':
+      case 'pgsql':
+      case 'sqlite':
+      case 'sqlsrv':
+        break;                    
+    }
+    
     if ($set) {
       $this->resetDb();
     }
